jhotdraw-core/src/main/java/org/jhotdraw/draw/figure/BezierFigure.java

```diff
@@ -171,11 +171,12 @@ protected void drawFill(Graphics2D g) {
 
   @Override
   public boolean contains(Point2D.Double p, double scaleDenominator) {
+    boolean result = false;
     double tolerance =
         Math.max(1f, 2 * AttributeKeys.getPerpendicularHitGrowth(this, scaleDenominator));
     if (isClosed() || attr().get(FILL_COLOR) != null && attr().get(UNCLOSED_PATH_FILLED)) {
       if (path.contains(p)) {
-        return true;
+        result = true;
```

Comment:
The idea of using a variable for the return and having only one line of code doing the return is a good idea. However, with the current implementation, the behaviour of the method will be different from the previous version. When the return statement is used, none of the code afterwards is executed. This is not the case with the current implementation. The lines 180-192 should be in an "else" statement so that it is not be executed if it goes in the first "if" statement.